From 543eef55043a975c2ede3e8d9457c1456f9186e4 Mon Sep 17 00:00:00 2001
From: Mark Vitale <mvitale@sinenomine.net>
Date: Wed, 23 Sep 2020 17:02:52 -0400
Subject: [PATCH 07/12] afs: remove vestigial externs for afs_xcbhash

Commit 64cc7f0ca7a44bb214396c829268a541ab286c69 "afs: Create
afs_StaleVCache" consolidated many references to afs_xcbhash into a new
function afs_StaleVCache.  However, this left many references to 'extern
afs_wrlock_t afs_xcbhash' that are no longer needed.

But actually, many of these have not been needed since
src/afs/afs_prototypes.h gained 'extern afs_rwlock_t afs_xcbhash' with
commit 8f2df21ffe59e9aa66219bf24656775b584c122d
"pull-prototypes-to-head-20020821"

Remove the vestigial extern references.

No functional change is incurred by this commit.

Change-Id: Ie6cfb6d90c52951795378d3b42e041567d207305
Reviewed-on: https://gerrit.openafs.org/14405
Reviewed-by: Andrew Deason <adeason@sinenomine.net>
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Benjamin Kaduk <kaduk@mit.edu>
(cherry picked from commit a3fc79633fb0601bf02508bd1e64652f403e4b7e)
---
 src/afs/VNOPS/afs_vnop_attrs.c   | 1 -
 src/afs/VNOPS/afs_vnop_dirops.c  | 1 -
 src/afs/VNOPS/afs_vnop_link.c    | 2 --
 src/afs/VNOPS/afs_vnop_remove.c  | 1 -
 src/afs/VNOPS/afs_vnop_rename.c  | 2 --
 src/afs/VNOPS/afs_vnop_symlink.c | 1 -
 src/afs/afs_bypasscache.c        | 2 --
 7 files changed, 10 deletions(-)

diff --git a/src/afs/VNOPS/afs_vnop_attrs.c b/src/afs/VNOPS/afs_vnop_attrs.c
index 645cf3c0a..0c2620ed0 100644
--- a/src/afs/VNOPS/afs_vnop_attrs.c
+++ b/src/afs/VNOPS/afs_vnop_attrs.c
@@ -33,7 +33,6 @@
 #include "afs/nfsclient.h"
 #include "afs/afs_osidnlc.h"
 
-extern afs_rwlock_t afs_xcbhash;
 struct afs_exporter *afs_nfsexporter;
 extern struct vcache *afs_globalVp;
 #if defined(AFS_HPUX110_ENV)
diff --git a/src/afs/VNOPS/afs_vnop_dirops.c b/src/afs/VNOPS/afs_vnop_dirops.c
index 596651324..4c67c4be4 100644
--- a/src/afs/VNOPS/afs_vnop_dirops.c
+++ b/src/afs/VNOPS/afs_vnop_dirops.c
@@ -29,7 +29,6 @@
 #include "afs/afs_osidnlc.h"
 
 extern afs_rwlock_t afs_xvcache;
-extern afs_rwlock_t afs_xcbhash;
 
 /* don't set CDirty in here because RPC is called synchronously */
 
diff --git a/src/afs/VNOPS/afs_vnop_link.c b/src/afs/VNOPS/afs_vnop_link.c
index f57b7ad49..c74aa36e7 100644
--- a/src/afs/VNOPS/afs_vnop_link.c
+++ b/src/afs/VNOPS/afs_vnop_link.c
@@ -24,8 +24,6 @@
 #include "afs/nfsclient.h"
 #include "afs/afs_osidnlc.h"
 
-extern afs_rwlock_t afs_xcbhash;
-
 /* Note that we don't set CDirty here, this is OK because the link
  * RPC is called synchronously. */
 
diff --git a/src/afs/VNOPS/afs_vnop_remove.c b/src/afs/VNOPS/afs_vnop_remove.c
index 33bc14218..eca9eed24 100644
--- a/src/afs/VNOPS/afs_vnop_remove.c
+++ b/src/afs/VNOPS/afs_vnop_remove.c
@@ -28,7 +28,6 @@
 
 
 extern afs_rwlock_t afs_xvcache;
-extern afs_rwlock_t afs_xcbhash;
 
 
 static void
diff --git a/src/afs/VNOPS/afs_vnop_rename.c b/src/afs/VNOPS/afs_vnop_rename.c
index 2d4b9c185..59a1d5032 100644
--- a/src/afs/VNOPS/afs_vnop_rename.c
+++ b/src/afs/VNOPS/afs_vnop_rename.c
@@ -25,8 +25,6 @@
 #include "afs/nfsclient.h"
 #include "afs/afs_osidnlc.h"
 
-extern afs_rwlock_t afs_xcbhash;
-
 /* Note that we don't set CDirty here, this is OK because the rename
  * RPC is called synchronously. */
 
diff --git a/src/afs/VNOPS/afs_vnop_symlink.c b/src/afs/VNOPS/afs_vnop_symlink.c
index e88e09ecc..c9be5f6b7 100644
--- a/src/afs/VNOPS/afs_vnop_symlink.c
+++ b/src/afs/VNOPS/afs_vnop_symlink.c
@@ -30,7 +30,6 @@
 #include "afs/afs_osidnlc.h"
 
 extern afs_rwlock_t afs_xvcache;
-extern afs_rwlock_t afs_xcbhash;
 
 /* Note: There is the bare bones beginning of symlink hints in the now
  * defunct afs/afs_lookup.c file. Since they are not in use, making the call
diff --git a/src/afs/afs_bypasscache.c b/src/afs/afs_bypasscache.c
index 64da1ed19..c51bd25c7 100644
--- a/src/afs/afs_bypasscache.c
+++ b/src/afs/afs_bypasscache.c
@@ -112,8 +112,6 @@ int cache_bypass_strategy   = 	NEVER_BYPASS_CACHE;
 afs_size_t cache_bypass_threshold  =  	AFS_CACHE_BYPASS_DISABLED; /* file size > threshold triggers bypass */
 int cache_bypass_prefetch = 1;	/* Should we do prefetching ? */
 
-extern afs_rwlock_t afs_xcbhash;
-
 /*
  * This is almost exactly like the PFlush() routine in afs_pioctl.c,
  * but that routine is static.  We are about to change a file from
-- 
2.36.1

